### PR TITLE
ISPN-14079 Jakarta dependencies depend on non-jakarta ones

### PR DESCRIPTION
**Pull request by jabolina**
https://issues.redhat.com/browse/ISPN-14079

As discussed in Zulip, I've created a new module for the artifacts adding the `-jakarta` postfix. Tristan helped listing the modules we use the eclipse converter and I've created a new module for each, with exception of `extended-statistics`, that even though is using the eclipse plugin it does not have any `javax` use. Should I create a new jakarta module for it too?

Also, comparing against the jar generated in main, the ones in this PR seems to have all the same stuff, but I dont know a good way that we can test it, so suggestions are pretty welcome.

### CODE REVIEW

**Comment by tristantarrant** [comment]
I've just tried to create a simple project with a single dependency (`org.infinispan:infinispan-client-hotrod-jakarta`) and unfortunately non-jakarta dependencies are still being pulled in:
```
INFO] org.infinispan.playground:hotrod-authentication:jar:0.0.1-SNAPSHOT
[INFO] +- org.infinispan:infinispan-client-hotrod-jakarta:jar:14.0.0-SNAPSHOT:compile
[INFO] |  +- org.infinispan:infinispan-client-hotrod:jar:14.0.0-SNAPSHOT:compile
```
Additionally it is depending on the legacy `jakarta.transaction:jakarta.transaction-api:jar:1.3.3:compile` instead of the new `jakarta.transaction:jakarta.transaction-api:jar:2.0.1:compile`

File: cdi/common-jakarta/pom.xml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    <parent>
+        <groupId>org.infinispan</groupId>
+        <artifactId>infinispan-cdi-parent</artifactId>
+        <version>14.0.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>infinispan-cdi-common-jakarta</artifactId>
+    <name>Infinispan CDI JakartaEE support common</name>
```
Comment (rhusar) [inline cdi/common-jakarta/pom.xml:15]: Kind of a nitpick but "JakartaEE" as one word is not a thing, use "Jakarta EE" instead. Applies to many places. Simple find & replace should do the trick.

**Review by rhusar — CHANGES_REQUESTED** [review]

**Comment by jabolina** [comment]
Updated to use a `provided` scope, but I am bit afraid that I am only delegating the need to pull the non-jakarta dependencies to something else. The tree now looks like this:

```text
[INFO] com.jabolina:playground:jar:1.0-SNAPSHOT
[INFO] \- org.infinispan:infinispan-client-hotrod-jakarta:jar:14.0.0-SNAPSHOT:compile
[INFO]    \- org.infinispan:infinispan-commons-jakarta:jar:14.0.0-SNAPSHOT:compile
```

In the last PR I've updated the transaction API, meaning that lots of places needed to update from `javax` to `jakarta`.
Also includes a change removing Geronimo stuff. Looks like there is no support for the transaction version we are using.

File: cdi/common-jakarta/pom.xml (hunk 1 above)
Comment (jabolina) [reply cdi/common-jakarta/pom.xml]: Updated, thanks!

File: cdi/embedded-jakarta/src/main/java/org/infinispan/cdi/embedded/package-info.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Version of Infinispan CDI Embedded using the JakartaEE APIs.
```
Comment (rhusar) [inline cdi/embedded-jakarta/src/main/java/org/infinispan/cdi/embedded/package-info.java:2]: @jabolina Let's fix these in Javadoc as well and use "Jakarta EE" instead (there are ~12 occurences).

**Review by rhusar — CHANGES_REQUESTED** [review]

**Comment by jabolina** [comment]
CI is back to normal. Should be good to go now.

**Comment by tristantarrant** [comment]
All of the jakarta modules need to have their correct `compile` dependencies added. For example, `hotrod-client-jakarta` should depend on the `netty` components, and so on.

**Comment by tristantarrant** [comment]
Additionally, the `cdi-jakarta` module should depend on `jakarta.enterprise.cdi-api` 4.0

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**